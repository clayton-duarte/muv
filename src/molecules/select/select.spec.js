import React from 'react';

import Select from '.';

it('should render a select', () => {
  const wrapper = mount(
    <Select />,
  );
  expect(wrapper).toMatchSnapshot();
});
