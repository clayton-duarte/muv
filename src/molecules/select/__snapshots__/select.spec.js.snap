// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a select 1`] = `
<Select
  onCancel={[Function]}
  onConfirm={[Function]}
  placeholder="Selecione..."
  render={
    Object {
      "data": Array [
        Object {
          "char": "a",
        },
        Object {
          "char": "b",
        },
      ],
      "title": Array [
        "a",
        "b",
      ],
    }
  }
>
  <styled.div
    theme={
      Object {
        "color": Object {
          "black": "#222222",
          "darkgray": "#515151",
          "error": "#dd3333",
          "gray": "#a9abae",
          "lightgray": "#e5e5e5",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0081c8",
          "secondary": "#00245f",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "font": Object {
          "color": "#222222",
          "family": "system-ui, sans-serif",
          "size": "16px",
          "spacing": 0,
        },
        "shape": Object {
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".75rem .875rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
          "size": "16px",
        },
        "transition": Object {
          "time": ".3s ease",
        },
      }
    }
  >
    <div
      className="sc-gqjmRU bDPTKv"
    >
      <Styled(Text)
        theme={
          Object {
            "color": Object {
              "black": "#222222",
              "darkgray": "#515151",
              "error": "#dd3333",
              "gray": "#a9abae",
              "lightgray": "#e5e5e5",
              "overlay": "rgba(0, 0, 0, .15)",
              "primary": "#0081c8",
              "secondary": "#00245f",
              "success": "#33bb33",
              "warn": "#dd3333",
              "white": "#fefefe",
            },
            "font": Object {
              "color": "#222222",
              "family": "system-ui, sans-serif",
              "size": "16px",
              "spacing": 0,
            },
            "shape": Object {
              "border": ".125rem solid rgba(0, 0, 0, .15)",
              "margin": ".25rem .125rem",
              "padding": ".75rem .875rem",
              "radius": ".25rem",
              "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
              "size": "16px",
            },
            "transition": Object {
              "time": ".3s ease",
            },
          }
        }
      >
        <Text
          align="left"
          className="sc-iwsKbI eUFIcZ"
          label={false}
          strong={false}
          theme={
            Object {
              "color": Object {
                "black": "#222222",
                "darkgray": "#515151",
                "error": "#dd3333",
                "gray": "#a9abae",
                "lightgray": "#e5e5e5",
                "overlay": "rgba(0, 0, 0, .15)",
                "primary": "#0081c8",
                "secondary": "#00245f",
                "success": "#33bb33",
                "warn": "#dd3333",
                "white": "#fefefe",
              },
              "font": Object {
                "color": "#222222",
                "family": "system-ui, sans-serif",
                "size": "16px",
                "spacing": 0,
              },
              "shape": Object {
                "border": ".125rem solid rgba(0, 0, 0, .15)",
                "margin": ".25rem .125rem",
                "padding": ".75rem .875rem",
                "radius": ".25rem",
                "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                "size": "16px",
              },
              "transition": Object {
                "time": ".3s ease",
              },
            }
          }
          transform="none"
          type=""
        >
          <styled.p
            align="left"
            className="sc-iwsKbI eUFIcZ"
            label={false}
            strong={false}
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "system-ui, sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".25rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
            transform="none"
          >
            <p
              className="sc-iwsKbI eUFIcZ sc-bdVaJa utCNp"
              label={false}
              transform="none"
            >
              Selecione...
            </p>
          </styled.p>
        </Text>
      </Styled(Text)>
      <styled.button
        length={2}
        onBlur={[Function]}
        onCancel={[Function]}
        onClick={[Function]}
        onConfirm={[Function]}
        placeholder="Selecione..."
        render={
          Object {
            "data": Array [
              Object {
                "char": "a",
              },
              Object {
                "char": "b",
              },
            ],
            "title": Array [
              "a",
              "b",
            ],
          }
        }
        selected={false}
        theme={
          Object {
            "color": Object {
              "black": "#222222",
              "darkgray": "#515151",
              "error": "#dd3333",
              "gray": "#a9abae",
              "lightgray": "#e5e5e5",
              "overlay": "rgba(0, 0, 0, .15)",
              "primary": "#0081c8",
              "secondary": "#00245f",
              "success": "#33bb33",
              "warn": "#dd3333",
              "white": "#fefefe",
            },
            "font": Object {
              "color": "#222222",
              "family": "system-ui, sans-serif",
              "size": "16px",
              "spacing": 0,
            },
            "shape": Object {
              "border": ".125rem solid rgba(0, 0, 0, .15)",
              "margin": ".25rem .125rem",
              "padding": ".75rem .875rem",
              "radius": ".25rem",
              "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
              "size": "16px",
            },
            "transition": Object {
              "time": ".3s ease",
            },
          }
        }
      >
        <button
          className="sc-dnqmqq jgpFmF"
          onBlur={[Function]}
          onClick={[Function]}
          placeholder="Selecione..."
          selected={false}
        >
          <Styled(Text)
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "system-ui, sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".25rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
          >
            <Text
              align="left"
              className="sc-iwsKbI eUFIcZ"
              label={false}
              strong={false}
              theme={
                Object {
                  "color": Object {
                    "black": "#222222",
                    "darkgray": "#515151",
                    "error": "#dd3333",
                    "gray": "#a9abae",
                    "lightgray": "#e5e5e5",
                    "overlay": "rgba(0, 0, 0, .15)",
                    "primary": "#0081c8",
                    "secondary": "#00245f",
                    "success": "#33bb33",
                    "warn": "#dd3333",
                    "white": "#fefefe",
                  },
                  "font": Object {
                    "color": "#222222",
                    "family": "system-ui, sans-serif",
                    "size": "16px",
                    "spacing": 0,
                  },
                  "shape": Object {
                    "border": ".125rem solid rgba(0, 0, 0, .15)",
                    "margin": ".25rem .125rem",
                    "padding": ".75rem .875rem",
                    "radius": ".25rem",
                    "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                    "size": "16px",
                  },
                  "transition": Object {
                    "time": ".3s ease",
                  },
                }
              }
              transform="none"
              type=""
            >
              <styled.p
                align="left"
                className="sc-iwsKbI eUFIcZ"
                label={false}
                strong={false}
                theme={
                  Object {
                    "color": Object {
                      "black": "#222222",
                      "darkgray": "#515151",
                      "error": "#dd3333",
                      "gray": "#a9abae",
                      "lightgray": "#e5e5e5",
                      "overlay": "rgba(0, 0, 0, .15)",
                      "primary": "#0081c8",
                      "secondary": "#00245f",
                      "success": "#33bb33",
                      "warn": "#dd3333",
                      "white": "#fefefe",
                    },
                    "font": Object {
                      "color": "#222222",
                      "family": "system-ui, sans-serif",
                      "size": "16px",
                      "spacing": 0,
                    },
                    "shape": Object {
                      "border": ".125rem solid rgba(0, 0, 0, .15)",
                      "margin": ".25rem .125rem",
                      "padding": ".75rem .875rem",
                      "radius": ".25rem",
                      "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                      "size": "16px",
                    },
                    "transition": Object {
                      "time": ".3s ease",
                    },
                  }
                }
                transform="none"
              >
                <p
                  className="sc-iwsKbI eUFIcZ sc-bdVaJa utCNp"
                  label={false}
                  transform="none"
                >
                  Selecione...
                </p>
              </styled.p>
            </Text>
          </Styled(Text)>
          <Styled(Text)
            key="a"
            onClick={[Function]}
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "system-ui, sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".25rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
          >
            <Text
              align="left"
              className="sc-iwsKbI eUFIcZ"
              label={false}
              onClick={[Function]}
              strong={false}
              theme={
                Object {
                  "color": Object {
                    "black": "#222222",
                    "darkgray": "#515151",
                    "error": "#dd3333",
                    "gray": "#a9abae",
                    "lightgray": "#e5e5e5",
                    "overlay": "rgba(0, 0, 0, .15)",
                    "primary": "#0081c8",
                    "secondary": "#00245f",
                    "success": "#33bb33",
                    "warn": "#dd3333",
                    "white": "#fefefe",
                  },
                  "font": Object {
                    "color": "#222222",
                    "family": "system-ui, sans-serif",
                    "size": "16px",
                    "spacing": 0,
                  },
                  "shape": Object {
                    "border": ".125rem solid rgba(0, 0, 0, .15)",
                    "margin": ".25rem .125rem",
                    "padding": ".75rem .875rem",
                    "radius": ".25rem",
                    "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                    "size": "16px",
                  },
                  "transition": Object {
                    "time": ".3s ease",
                  },
                }
              }
              transform="none"
              type=""
            >
              <styled.p
                align="left"
                className="sc-iwsKbI eUFIcZ"
                label={false}
                onClick={[Function]}
                strong={false}
                theme={
                  Object {
                    "color": Object {
                      "black": "#222222",
                      "darkgray": "#515151",
                      "error": "#dd3333",
                      "gray": "#a9abae",
                      "lightgray": "#e5e5e5",
                      "overlay": "rgba(0, 0, 0, .15)",
                      "primary": "#0081c8",
                      "secondary": "#00245f",
                      "success": "#33bb33",
                      "warn": "#dd3333",
                      "white": "#fefefe",
                    },
                    "font": Object {
                      "color": "#222222",
                      "family": "system-ui, sans-serif",
                      "size": "16px",
                      "spacing": 0,
                    },
                    "shape": Object {
                      "border": ".125rem solid rgba(0, 0, 0, .15)",
                      "margin": ".25rem .125rem",
                      "padding": ".75rem .875rem",
                      "radius": ".25rem",
                      "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                      "size": "16px",
                    },
                    "transition": Object {
                      "time": ".3s ease",
                    },
                  }
                }
                transform="none"
              >
                <p
                  className="sc-iwsKbI eUFIcZ sc-bdVaJa utCNp"
                  label={false}
                  onClick={[Function]}
                  transform="none"
                >
                  a
                </p>
              </styled.p>
            </Text>
          </Styled(Text)>
          <Styled(Text)
            key="b"
            onClick={[Function]}
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "system-ui, sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".25rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
          >
            <Text
              align="left"
              className="sc-iwsKbI eUFIcZ"
              label={false}
              onClick={[Function]}
              strong={false}
              theme={
                Object {
                  "color": Object {
                    "black": "#222222",
                    "darkgray": "#515151",
                    "error": "#dd3333",
                    "gray": "#a9abae",
                    "lightgray": "#e5e5e5",
                    "overlay": "rgba(0, 0, 0, .15)",
                    "primary": "#0081c8",
                    "secondary": "#00245f",
                    "success": "#33bb33",
                    "warn": "#dd3333",
                    "white": "#fefefe",
                  },
                  "font": Object {
                    "color": "#222222",
                    "family": "system-ui, sans-serif",
                    "size": "16px",
                    "spacing": 0,
                  },
                  "shape": Object {
                    "border": ".125rem solid rgba(0, 0, 0, .15)",
                    "margin": ".25rem .125rem",
                    "padding": ".75rem .875rem",
                    "radius": ".25rem",
                    "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                    "size": "16px",
                  },
                  "transition": Object {
                    "time": ".3s ease",
                  },
                }
              }
              transform="none"
              type=""
            >
              <styled.p
                align="left"
                className="sc-iwsKbI eUFIcZ"
                label={false}
                onClick={[Function]}
                strong={false}
                theme={
                  Object {
                    "color": Object {
                      "black": "#222222",
                      "darkgray": "#515151",
                      "error": "#dd3333",
                      "gray": "#a9abae",
                      "lightgray": "#e5e5e5",
                      "overlay": "rgba(0, 0, 0, .15)",
                      "primary": "#0081c8",
                      "secondary": "#00245f",
                      "success": "#33bb33",
                      "warn": "#dd3333",
                      "white": "#fefefe",
                    },
                    "font": Object {
                      "color": "#222222",
                      "family": "system-ui, sans-serif",
                      "size": "16px",
                      "spacing": 0,
                    },
                    "shape": Object {
                      "border": ".125rem solid rgba(0, 0, 0, .15)",
                      "margin": ".25rem .125rem",
                      "padding": ".75rem .875rem",
                      "radius": ".25rem",
                      "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                      "size": "16px",
                    },
                    "transition": Object {
                      "time": ".3s ease",
                    },
                  }
                }
                transform="none"
              >
                <p
                  className="sc-iwsKbI eUFIcZ sc-bdVaJa utCNp"
                  label={false}
                  onClick={[Function]}
                  transform="none"
                >
                  b
                </p>
              </styled.p>
            </Text>
          </Styled(Text)>
          <Styled(Arrow)
            theme={
              Object {
                "color": Object {
                  "black": "#222222",
                  "darkgray": "#515151",
                  "error": "#dd3333",
                  "gray": "#a9abae",
                  "lightgray": "#e5e5e5",
                  "overlay": "rgba(0, 0, 0, .15)",
                  "primary": "#0081c8",
                  "secondary": "#00245f",
                  "success": "#33bb33",
                  "warn": "#dd3333",
                  "white": "#fefefe",
                },
                "font": Object {
                  "color": "#222222",
                  "family": "system-ui, sans-serif",
                  "size": "16px",
                  "spacing": 0,
                },
                "shape": Object {
                  "border": ".125rem solid rgba(0, 0, 0, .15)",
                  "margin": ".25rem .125rem",
                  "padding": ".75rem .875rem",
                  "radius": ".25rem",
                  "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                  "size": "16px",
                },
                "transition": Object {
                  "time": ".3s ease",
                },
              }
            }
          >
            <Arrow
              className="sc-gZMcBi bNtIrL"
              theme={
                Object {
                  "color": Object {
                    "black": "#222222",
                    "darkgray": "#515151",
                    "error": "#dd3333",
                    "gray": "#a9abae",
                    "lightgray": "#e5e5e5",
                    "overlay": "rgba(0, 0, 0, .15)",
                    "primary": "#0081c8",
                    "secondary": "#00245f",
                    "success": "#33bb33",
                    "warn": "#dd3333",
                    "white": "#fefefe",
                  },
                  "font": Object {
                    "color": "#222222",
                    "family": "system-ui, sans-serif",
                    "size": "16px",
                    "spacing": 0,
                  },
                  "shape": Object {
                    "border": ".125rem solid rgba(0, 0, 0, .15)",
                    "margin": ".25rem .125rem",
                    "padding": ".75rem .875rem",
                    "radius": ".25rem",
                    "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                    "size": "16px",
                  },
                  "transition": Object {
                    "time": ".3s ease",
                  },
                }
              }
              up={false}
            >
              <Styled(Text)
                className="sc-gZMcBi bNtIrL"
                theme={
                  Object {
                    "color": Object {
                      "black": "#222222",
                      "darkgray": "#515151",
                      "error": "#dd3333",
                      "gray": "#a9abae",
                      "lightgray": "#e5e5e5",
                      "overlay": "rgba(0, 0, 0, .15)",
                      "primary": "#0081c8",
                      "secondary": "#00245f",
                      "success": "#33bb33",
                      "warn": "#dd3333",
                      "white": "#fefefe",
                    },
                    "font": Object {
                      "color": "#222222",
                      "family": "system-ui, sans-serif",
                      "size": "16px",
                      "spacing": 0,
                    },
                    "shape": Object {
                      "border": ".125rem solid rgba(0, 0, 0, .15)",
                      "margin": ".25rem .125rem",
                      "padding": ".75rem .875rem",
                      "radius": ".25rem",
                      "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                      "size": "16px",
                    },
                    "transition": Object {
                      "time": ".3s ease",
                    },
                  }
                }
                type="span"
                up={false}
              >
                <Text
                  align="left"
                  className="sc-gZMcBi bNtIrL sc-htoDjs hEAdvI"
                  label={false}
                  strong={false}
                  theme={
                    Object {
                      "color": Object {
                        "black": "#222222",
                        "darkgray": "#515151",
                        "error": "#dd3333",
                        "gray": "#a9abae",
                        "lightgray": "#e5e5e5",
                        "overlay": "rgba(0, 0, 0, .15)",
                        "primary": "#0081c8",
                        "secondary": "#00245f",
                        "success": "#33bb33",
                        "warn": "#dd3333",
                        "white": "#fefefe",
                      },
                      "font": Object {
                        "color": "#222222",
                        "family": "system-ui, sans-serif",
                        "size": "16px",
                        "spacing": 0,
                      },
                      "shape": Object {
                        "border": ".125rem solid rgba(0, 0, 0, .15)",
                        "margin": ".25rem .125rem",
                        "padding": ".75rem .875rem",
                        "radius": ".25rem",
                        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                        "size": "16px",
                      },
                      "transition": Object {
                        "time": ".3s ease",
                      },
                    }
                  }
                  transform="none"
                  type="span"
                  up={false}
                >
                  <styled.span
                    align="left"
                    className="sc-gZMcBi bNtIrL sc-htoDjs hEAdvI"
                    label={false}
                    strong={false}
                    theme={
                      Object {
                        "color": Object {
                          "black": "#222222",
                          "darkgray": "#515151",
                          "error": "#dd3333",
                          "gray": "#a9abae",
                          "lightgray": "#e5e5e5",
                          "overlay": "rgba(0, 0, 0, .15)",
                          "primary": "#0081c8",
                          "secondary": "#00245f",
                          "success": "#33bb33",
                          "warn": "#dd3333",
                          "white": "#fefefe",
                        },
                        "font": Object {
                          "color": "#222222",
                          "family": "system-ui, sans-serif",
                          "size": "16px",
                          "spacing": 0,
                        },
                        "shape": Object {
                          "border": ".125rem solid rgba(0, 0, 0, .15)",
                          "margin": ".25rem .125rem",
                          "padding": ".75rem .875rem",
                          "radius": ".25rem",
                          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
                          "size": "16px",
                        },
                        "transition": Object {
                          "time": ".3s ease",
                        },
                      }
                    }
                    transform="none"
                    up={false}
                  >
                    <span
                      className="sc-gZMcBi bNtIrL sc-htoDjs hEAdvI sc-bwzfXH fCnCPt"
                      label={false}
                      transform="none"
                    >
                      ◄
                    </span>
                  </styled.span>
                </Text>
              </Styled(Text)>
            </Arrow>
          </Styled(Arrow)>
        </button>
      </styled.button>
    </div>
  </styled.div>
</Select>
`;
